### PR TITLE
[MM-45616] add insights as last channel viewed

### PR DESCRIPTION
**Pull request by Willyfrog**
#### Summary
Make insights available as last channel viewed

it is missing the tests, but I'll try to get them tomorrow, in the meantime you can review.

#### Screenshots

Screenshot 2022-06-23 at 19 36 22

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by mkraft — APPROVED** [review]
Code LGTM.

**Comment by mkraft** [comment]
@jfrerich I hope you don't mind I swapped you into this review instead of Ben because he's on PTO until next week.

**Comment by Willyfrog** [comment]
> @Willyfrog Just so that I'm sure I understand the purpose... this will make you return to insights if it was the last thing you were looking at prior to changing teams, right?

Exactly

**Comment by Willyfrog** [comment]
@saturninoabril I added the check for the feature flag in cypress test, is that ok? I've been trying to figure it out and seems like this is the expected approach

**Comment by Willyfrog** [comment]
Thanks for confirming, I was not really sure as the test fails if the featureflag is not set

**Review by jfrerich — APPROVED** [review]
LGTM!  Nice work @Willyfrog

**Comment by jgilliam17** [comment]
@Willyfrog Can you please resolve conflicts and update so we can trigger e2es? Thanks 🙂

File: e2e/cypress/tests/integration/insights/last_viewed_spec.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        // # Go to the Insights view on Team A
+        cy.uiGetSidebarInsightsButton().click();
+
+        // # Switch to Test Channel
+        cy.uiClickSidebarItem(testChannel.name);
+
+        // # Leave the current channel
+        cy.uiLeaveChannel();
+
+        // Verify url is set up for threads view when thread view is the penultimate view
```
Comment (jgilliam17) [inline e2e/cypress/tests/integration/insights/last_viewed_spec.js:76]:
```suggestion
        // * Verify url is set up for insights view when insights view is the penultimate view
```

**Review by jgilliam17 — CHANGES_REQUESTED** [review]
@Willyfrog 
I added test IDs and few suggestions to update descriptions
I'll trigger the e2es after we add these - thanks

**Comment by Willyfrog** [comment]
@jgilliam17 added your suggestions (thanks!) and solved the conflict

**Review by jgilliam17 — APPROVED** [review]
Thanks @Willyfrog 
Tested, looks good to merge
Verified insights are available as last channel viewed.
E2E report - no PR related failures. New e2es added here are passing.
Screen Shot 2022-07-08 at 11 40 53 AM

**Comment by jgilliam17** [comment]
Added Jira ticket for release testing https://mattermost.atlassian.net/browse/MM-45616

**Pull request merged**